### PR TITLE
Replace filter interface

### LINKED ISSUE #108: Extra filter interface for plugin
**What would you like to be added**:
Extra filter interface for plugin
**Why is this needed**:
I extracted Filter interface ritems to a project https://github.com/dubbogo/dubbo-go-proxy-filter.
Any Suggestion?

### PR DESCRIPTION
**Pull request by mark4z**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #108 

**Special notes for your reviewer**:

manual： https://www.yuque.com/dubbogoproxy/vld1hq/kgauze
**check it and maybe help understand it.**

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by cityiron** [comment]
Tks, we will review next days.

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,7 @@ go 1.14
 require (
 	github.com/apache/dubbo-go v1.5.5
 	github.com/apache/dubbo-go-hessian2 v1.7.0
+	github.com/dubbogo/dubbo-go-proxy-filter v0.1.0-rc1.0.20210120132524-c63f4eb13725 //TODO
 	github.com/dubbogo/go-zookeeper v1.0.2
 	github.com/emirpasic/gods v1.12.0
```
Comment (xiaoliu10) [inline go.mod:10]: split import

File: go.sum [hunk 2]
```diff
@@ -225,6 +225,8 @@ github.com/docker/go-connections v0.4.0 h1:El9xVISelRB7BuFusrZozjnkIM5YnzCViNKoh
 github.com/docker/go-connections v0.4.0/go.mod h1:Gbd7IOopHjR8Iph03tsViu4nIes5XhDvyHbTtUxmeec=
 github.com/docker/go-units v0.4.0/go.mod h1:fgPhTUdO+D/Jk86RDLlptpiXQzgHJF7gydDDbaIK4Dk=
 github.com/docker/spdystream v0.0.0-20160310174837-449fdfce4d96/go.mod h1:Qh8CwZgvJUkLughtfhJv5dyTYa91l1fOUCrgjqmcifM=
```
Comment (xiaoliu10) [inline go.sum:227]: split import

File: go.mod (hunk 1 above)
Comment (mark4z) [reply go.mod:10]: If every thing ok I will release the `dubbo-go-proxy-filter` and update this.

File: pkg/client/request.go [hunk 3]
```diff
@@ -23,8 +23,8 @@ import (
 )
 
 import (
-	"github.com/dubbogo/dubbo-go-proxy/pkg/config"
-	"github.com/dubbogo/dubbo-go-proxy/pkg/router"
+	"github.com/dubbogo/dubbo-go-proxy-filter/pkg/api/config"
```
Comment (xiaoliu10) [inline pkg/client/request.go:26]: split import

**Review by xiaoliu10 — APPROVED** [review]

**Review by williamfeng323 — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**